pkg/ccl/backupccl/restore_job.go

```diff
@@ -657,15 +660,32 @@ func shouldPreRestore(table *tabledesc.Mutable) bool {
 	return ok
 }
 
-// createImportingDescriptors create the tables that we will restore into. It also
-// fetches the information from the old tables that we need for the restore.
+// createImportingDescriptors create the tables that we will restore into and returns up to three
+// configurations for separate restoration flows. The three restoration flows are
+//
+//   1. dataToPreRestore: a restoration flow cfg to ingest a subset of
+//   system tables (e.g. zone configs) during a cluster restore that are
+//   required to be set up before the rest of the data gets restored.
+//   This should be empty during non-cluster restores.
+//
+//   2. preValidation: a restoration flow cfg to ingest the remainder of system tables,
```

Comment:
there's another split we could do here isn't there:
preRestore
mainRestore
postRestore

In cluster restore, system tables go into either pre or post, but never into main. Pre- and Post- are never validation-only; they're either empty or are real restores of system tables (including in schema-only).
Main is all requested tables, or all non-system table for cluster restore. It can be validation-only.

wdyt?

Reply:
Your setup seems to reason about compared to mine, but it implies that an additional flow would occur in cluster level, regular restore. Is there significant overhead required to setup an additional flow? If not, I'll go with your proposal.

Reply:
It also means that restores of _all_ system tables go untracked. Only one flow can get check-pointed. Is that fine?

If that postRestore flow takes a significant amount of time, the user may get worried about being stuck at 99% completion.

Reply:
ah, okay, yeah, your way seems better then.